asq-exercise-progress.html

```diff
@@ -0,0 +1,310 @@
+      },
+
+      submissionsNum: {
+        type: Number,
+        value: 0,
+      },
+
+      _submissionCounter: {
+        type: Number,
+        value: 0,
+      },
+
+      _workingUsersNum:{
+        type: Number,
+        value: 0,
+        computed: '_computeWorkingUsersNum(_users)',
+      },
+
+      _workingCounter: {
+        type: Number,
+        value: 0,
+      },
+
+     _focusedUsersNum:{
+        type: Number,
+        value: 0,
+        computed: '_computeFocusedUsersNum(_users)',
+     },
+
+     _focusedCounter: {
+       type: Number,
+       value: 0,
+     },
+
+     _idleUsersNum:{
+        type: Number,
+        value: 0,
+        computed: '_computeIdleUsersNum(_users)',
+     },
+
+     _idleCounter: {
+       type: Number,
+       value: 0,
+     },
+
+    },
+
+    _computeWorkingUsersNum: function() {
+      let users = this._users;
+      let num = Object.keys(users).reduce(function(acc, userId) {
+        if(users[userId].working) return ++acc;
+        return acc;
+      }, 0);
+      return num;
+    },
+
+    _computeFocusedUsersNum: function() {
+      let users = this._users;
+      return Object.keys(users).reduce(function(acc, userId) {
+        if(users[userId].focused && !users[userId].idle) return ++acc;
+        return acc;
+      }, 0);
+    },
+
+    _computeIdleUsersNum: function() {
+      let users = this._users;
+      return Object.keys(users).reduce(function(acc, userId) {
+        if(users[userId].idle || users[userId].disconnected) return ++acc
+        return acc;
+      }, 0);
+    },
+
+    _calcFlex: function(num) {
+      return "flex:" + num + ";";
+    },
+
+    _makeSureUserExists: function(userToFind) {
+      this._users[userToFind.id] = this._users[userToFind.id] || userToFind;
+      this._notifyPolymer();
+    },
+
+    _userWorked: function(user) {
+      this._makeSureUserExists(user);
+      if (!this._users[user.id].working) this._workingCounter++;
+      this._resetUser(user);
+      this._users[user.id].working = true;
+      this._notifyPolymer();
+    },
+
+    _userFocused: function(user) {
+      this._makeSureUserExists(user);
+      if (this._users[user.id].submitted) return;
+      if (!this._users[user.id].focused) this._focusedCounter++;
+      this._resetUser(user);
+      this._users[user.id].focused = true;
+      this._notifyPolymer();
+    },
+
+    _userMadeTabVisible: function(user) {
+      this._makeSureUserExists(user);
+      this._users[user.id].idle = false;
+      this._notifyPolymer();
+    },
+
+    _userIdled: function(user) {
+      this._makeSureUserExists(user);
+      if (this._users[user.id].submitted) return;
+      if (!this._users[user.id].idle) this._idleCounter++;
+      this._resetUser(user);
+      this._users[user.id].idle = true;
+      this._notifyPolymer();
+    },
+
+    _userMadeTabHidden: function(user) {
+      this._makeSureUserExists(user);
+      this._users[user.id].idle = false;
+      this._notifyPolymer();
```

Comment:
I tried using the two suggested methods in the documentation (i.e using the polymer set function or the notify-path function), but it didn't seem to work, maybe because the parameters have to be a concatenated string, since I need to use variables in there.